snuba/snuba_migrations/metrics/0035_metrics_add_prod_filter.py

```diff
@@ -0,0 +1,94 @@
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        # redo 0027
+        counters_op = get_forward_view_migration_polymorphic_table_v3(
+            source_table_name=self.raw_table_name,
+            table_name=self.counters_table_name,
+            aggregation_col_schema=[
+                Column("value", AggregateFunction("sum", [Float(64)])),
+            ],
+            mv_name=get_polymorphic_mv_variant_name("counters", self.mv_version),
+            aggregation_states="sumState(count_value) as value",
+            metric_type="counter",
+            target_mat_version=self.mv_version,
+            appended_where_clause="AND timestamp >= toDateTime('2022-03-30 00:11:00')",
+        )
+        counters_op.target = operations.OperationTarget.LOCAL
+
+        # redo 0032
+        distributions_op = get_forward_view_migration_polymorphic_table_v3(
+            source_table_name=self.raw_table_name,
+            table_name=self.dist_table_name,
+            aggregation_col_schema=COL_SCHEMA_DISTRIBUTIONS_V2,
+            mv_name=get_polymorphic_mv_variant_name("distributions", self.mv_version),
+            aggregation_states=(
+                "quantilesState(0.5, 0.75, 0.9, 0.95, 0.99)((arrayJoin(distribution_values) AS values_rows)) as percentiles, "
+                "minState(values_rows) as min, "
+                "maxState(values_rows) as max, "
+                "avgState(values_rows) as avg, "
+                "sumState(values_rows) as sum, "
+                "countState(values_rows) as count, "
+                "histogramState(250)(values_rows) as histogram_buckets"
+            ),
+            metric_type="distribution",
+            target_mat_version=self.mv_version,
+            appended_where_clause="AND timestamp >= toDateTime('2022-03-30 00:11:00')",
+        )
+        distributions_op.target = operations.OperationTarget.LOCAL
+        sets_op = get_forward_view_migration_polymorphic_table_v3(
+            source_table_name=self.raw_table_name,
+            table_name=self.sets_table_name,
+            aggregation_col_schema=[
+                Column("value", AggregateFunction("uniqCombined64", [UInt(64)])),
+            ],
+            aggregation_states="uniqCombined64State(arrayJoin(set_values)) as value",
+            mv_name=get_polymorphic_mv_variant_name("sets", self.mv_version),
+            metric_type="set",
+            target_mat_version=self.mv_version,
+            appended_where_clause="AND timestamp >= toDateTime('2022-03-30 00:11:00')",
+        )
+        sets_op.target = operations.OperationTarget.LOCAL
+
+        return [
+            counters_op,
+            operations.DropTable(
+                storage_set=StorageSetKey.METRICS,
+                table_name=get_polymorphic_mv_variant_name(
+                    "distributions", self.mv_version
+                ),
+                target=operations.OperationTarget.LOCAL,
+            ),
+            distributions_op,
```

Comment:
The materialized view version column is used differently. 
Clickhouse cannot replace a materialized view in an atomic manner (though newer version allow to mutate the statement, we should try that out). 
So we were used to create a first a new materialized view with a new version, then switch the version in the consumer so that we would start ingesting through the new matview, then we deleted the old matview.
This means the creation of the new matview must happen before you delete the old one and in two different migrations as the change to the consumer has to happen in between.

Details
https://www.notion.so/sentry/Migrations-vs-Materialized-Views-896bce040cb74a8fa9cca2e7c9a859ab

Probably a better option in new clickhosue versions:
https://clickhouse.com/docs/en/sql-reference/statements/alter/view/#alter-live-view-statement